packages/network/test/integration/proxy-connections.test.ts

```diff
@@ -68,93 +73,82 @@ describe('Proxy connection tests', () => {
             tracker?.stop(),
             onewayNode?.stop(),
             contactNode?.stop(),
-            contactNode2?.stop(),
+            contactNode2?.stop()
         ])
     })
 
     it('publisher node can form proxy connections', async () => {
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH, 'publisher')
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node-2', ProxyDirection.PUBLISH, 'publisher')
+        await onewayNode.setProxies(streamPartId, ['contact-node', 'contact-node-2'], ProxyDirection.PUBLISH, async () => 'publisher')
         // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(defaultStreamPartId)).toContainValues(['contact-node', 'contact-node-2'])
+        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(streamPartId)).toContainValues(['contact-node', 'contact-node-2'])
         // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(defaultStreamPartId)).toBeArrayOfSize(0)
+        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(streamPartId)).toBeArrayOfSize(0)
     })
-
+    
     it('subscriber node can form proxy connections', async () => {
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.SUBSCRIBE, 'subscriber')
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node-2', ProxyDirection.SUBSCRIBE, 'subscriber')
+        await onewayNode.setProxies(streamPartId, ['contact-node', 'contact-node-2'], ProxyDirection.SUBSCRIBE, async () => 'subscriber')
         // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getInboundNodesForStreamPart(defaultStreamPartId)).toContainValues(['contact-node', 'contact-node-2'])
+        expect(onewayNode.streamPartManager.getInboundNodesForStreamPart(streamPartId)).toContainValues(['contact-node', 'contact-node-2'])
         // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(defaultStreamPartId)).toBeArrayOfSize(0)
+        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(streamPartId)).toBeArrayOfSize(0)
     })
 
     it('publisher node can close proxy connections', async () => {
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node-2', ProxyDirection.PUBLISH, 'publisher')
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH, 'publisher')
+        await onewayNode.setProxies(streamPartId, ['contact-node', 'contact-node-2'], ProxyDirection.PUBLISH, async () => 'publisher')
```

Comment:
User ids are not relevant in these tests. Maybe we could hide that aspect into a helper function:
```
const getUserId = async () => 'mock-user-id'
```

```suggestion
        await onewayNode.setProxies(streamPartId, ['contact-node', 'contact-node-2'], ProxyDirection.PUBLISH, getUserId)
```